lang/cs/Org.Apache.REEF.IO/FileSystem/AzureDataLake/AzureDataLakeFileSystem.cs

```diff
@@ -0,0 +1,230 @@
+        /// </summary>
+        /// <exception cref="IOException">If directory cannot be created</exception>
+        public void CreateDirectory(Uri directoryUri)
+        {
+            bool createDirStatus = _adlsClient.CreateDirectory(directoryUri.AbsolutePath);
+            if (!createDirStatus)
+            {
+                throw new IOException($"Cannot create directory specified by {directoryUri}");
+            }
+        }
+
+        /// <summary>
+        /// Deletes a directory.
+        /// </summary>
+        /// <exception cref="IOException">If directory cannot be deleted</exception>
+        public void DeleteDirectory(Uri directoryUri)
+        {
+            bool deleteStatus = Exists(directoryUri) && 
+                _adlsClient.GetDirectoryEntry(directoryUri.AbsolutePath).Type == DirectoryEntryType.DIRECTORY &&
+                _adlsClient.DeleteRecursive(directoryUri.AbsolutePath);
+            if (!deleteStatus)
+            {
+                throw new IOException($"Cannot delete directory specified by {directoryUri}");
+            }
+        }
+
+        /// <summary>
+        /// Get the children on the given URI, if that refers to a directory.
+        /// </summary>
+        /// <exception cref="IOException">If directory does not exist</exception>
+        public IEnumerable<Uri> GetChildren(Uri directoryUri)
+        {
+            if (!Exists(directoryUri) || _adlsClient.GetDirectoryEntry(directoryUri.AbsolutePath).Type != DirectoryEntryType.DIRECTORY)
+            {
+                throw new IOException($"Cannot find directory specified by {directoryUri}");
+            }
+
+            return _adlsClient.EnumerateDirectory(directoryUri.AbsolutePath).Select(entry => new Uri($"{GetUriPrefix()}{entry.FullName}"));
+        }
+
+        /// <summary>
+        /// Create Uri from a given file path.
+        /// The file path can be full with prefix or relative without prefix.
+        /// If null is passed as the path, ArgumentException will be thrown.
+        /// </summary>
+        /// <exception cref="ArgumentNullException">If specified path is null</exception>
+        public Uri CreateUriForPath(string path)
+        {
+            if (path == null)
+            {
+                throw new ArgumentNullException(nameof(path), "Specified path is null");
+            }
+            return new Uri($"{GetUriPrefix()}/{path.TrimStart('/')}");
+        }
+
+        /// <summary>
+        /// Gets the FileStatus for remote file.
+        /// </summary>
+        /// <exception cref="ArgumentNullException">If remote file URI is null</exception>
+        /// <returns>FileStatus</returns>
+        public FileStatus GetFileStatus(Uri remoteFileUri)
+        {
+            if (remoteFileUri == null)
+            {
+                throw new ArgumentNullException(nameof(remoteFileUri), "Specified uri is null");
+            }
+            var entrySummary = _adlsClient.GetDirectoryEntry(remoteFileUri.AbsolutePath);
```

Comment:
Is it for a file or a directory?

Reply:
for both, whatever is specified by the "remoteFileUri.AbsolutePath"